src/Eshopworld.Messaging/Core/IMessenger.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Threading.Tasks;
+using Eshopworld.Core;
+
+namespace Eshopworld.Messaging.Core
+{
+    /// <summary>
+    /// Contract that provides a simple way to send events to topics
+    /// <remarks>
+    /// Custom topic name needs to be provided.
+    /// </remarks>
+    /// </summary>
+    public interface IPublishEventsWithTopicName: IDisposable
+    {
+        /// <summary>
+        /// Sends an event onto a topic.
+        /// </summary>
+        /// <typeparam name="T">The type of the event that we are sending.</typeparam>
+        /// <param name="event">The event that we are sending.</param>
+        /// <param name="topicName">The name of the topic we are sending to</param>
+        Task Publish<T>(T @event, string topicName) where T : class;
+    }
+
+    /// <summary>
+    /// Contract that provides a simple way to send and receive events through callbacks.
+    /// <remarks>
+    /// Custom topic name needs to be provided.
+    /// </remarks>
+    /// </summary>
+    public interface IDoPubSubWithTopicName: IPublishEventsWithTopicName, IMessageOperations
```

Comment:
Bit unclear on the naming convention and interface separation here :) Could be this is linked to existing package but maybe this is something we can start addressing as part of this work.

I see following interfaces that may be needed (names for discussion):
- IMessageSubscriber - subscribe to topics or queues, calls a handler you provide
- IMessagePublisher - pushes message to a queue or topic

I don't believe we need unified interface for sending and subscribing as these are independent concerns. You may have shared implementation behind but pushing them into the same interface seems unnecessary. The app/code will look much cleaner if you have them separated.

Reply:
in the existing package you have two main interfaces:
- `IDoFullMessaging: IDoMessages, IDoPubSub ` -> Send and subscribe to topic and queues using callbacks
- `IDoFullReactiveMessaging: IDoReactiveMessages, IDoReactiveEvents ` -> Send and subscribe to topic and queues using Rx 

I didn't want to pollute the existing interface and hence I created the new one. 
I think you are aligned to what @C-M-Ryan pointed out. 
I should be extending `IDoPubSub` to have an override that accepts the topic name

Reply:
updated here => https://github.com/eShopWorld/core/pull/31